### DISCUSSION TITLE
The 'vsPackage' package did not load correctly

### DISCUSSION THREAD

**Discussion by IADTIL** [dotnet/ResXResourceManager#470] [General]
Today Visual Studio suddenly started displaying an error, "**The 'vsPackage' package did not load correctly**".

This was being displayed every time I restarted Visual Studio, even after completely cleaning my project.

The ActivityLog.xml had this detail:

> CreateInstance failed for package [VsPackage]Source: 'mscorlib' Description: Exception has been thrown by the target of an invocation. System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.IO.FileNotFoundException: Could not load file or assembly 'TomsToolbox.Composition, Version=2.7.3.0, Culture=neutral, PublicKeyToken=60b39f873a8e3fc2' or one of its dependencies. The system cannot find the file specified.at ResXManager.VSIX.VsPackage..ctor()--- 

I noticed that ResXResourceManager had updated itself today (**4th April 2022**).  It is now at version **1.64.5123.0**

I disabled ResXResourceManager (In Visual Studio, go to Extensions->Manage Extensions), restarted Visual Studio with the same solution, and the error had gone.
Reenabled ResXResourceManager and restarted again and the error is still no longer displayed.

So, this is a heads up to:

**Users of this extension - If you see this problem, disabling and reenabling the extension seems to solve the problem.**

**Developers of this extension - Please could you investigate in case there is an underlying problem that this "fix" doesn't actually solve.**

Regards,
Ian Davidson.

PS.  I've put this as a "discussion", because as you can see I'm not sure if it is an "issue".

**Comment by tom-englert** [comment]
This is a well-known issue with the VSIX-Installer: https://github.com/dotnet/ResXResourceManager/blob/master/Documentation/Topics/Troubleshooting.md

**Reply by IADTIL** [reply]
Thank you!

Disabling and enabling it, as I mentioned, seems to have solved the problem, so one might not need to go as far as reinstalling it as per that troubleshooting note.  You might want to update it to suggest trying that first.   :) 

Cheers,
Ian.
